### PR TITLE
Resources supplier permission

### PR DESCRIPTION
**Pull request by NahuelValencia-eb**
Suppliers only are able to access to his resources

### CODE REVIEW

File: supplier_app/tests/test_view.py [hunk 1]
```diff
@@ -739,15 +748,24 @@ def test_taxpayer_detail_view_as_supplier_dont_show_approve_denied_btn(self):
...
+        self.ap_user.groups.add(self.ap_group)
+        self.client.force_login(self.ap_user)
+
+        self.supplier_group = Group.objects.get(name="supplier")
+        self.supplier_user = UserFactory(email='nahuelSupplier@gmail.com')
+        self.supplier_user.groups.add(self.supplier_group)
+        self.client2.force_login(self.supplier_user)
 
         self.taxpayer = TaxPayerArgentinaFactory()
+        self.comapnyuserpermission = CompanyUserPermissionFactory(
```
Comment (leandro-eb) [inline supplier_app/tests/test_view.py]: Please replace `self.comapnyuserpermission` by `self.company_user_permission`. Also consider thinking if we are talking about company permission, or user inside company permission, or company permission, to be clear about what are we talking about.

File: supplier_app/views.py [hunk 2]
```diff
@@ -199,58 +215,73 @@ def get_context_data(self, **kwargs):
...
+    def handle_no_permission(self):
+        return HttpResponseRedirect(Http404)
+
     def get_success_url(self, **kwargs):
         taxpayer_id = Address.objects.get(pk=self.kwargs['address_id']).taxpayer.id
         return reverse('supplier-details', kwargs={'taxpayer_id': taxpayer_id})
 
 
-class EditBankAccountView(UserLoginPermissionRequiredMixin, UpdateView):
+class EditBankAccountView(UserLoginPermissionRequiredMixin, HasTaxPayerPermissionMixin, UpdateView):
```
Comment (leandro-eb) [inline supplier_app/views.py]: Seems little weird to have `HasTaxPayerPermissionMixin`, I think if we are talking about a Mixin, it should be only `TaxPayerPermissionMixin` or something similar. Please contact me if you have doubts

File: supplier_app/tests/test_view.py (hunk 1 above)
Comment (gmarchetta-eb) [inline supplier_app/tests/test_view.py]: Typo on variable name

File: supplier_app/views.py [hunk 3]
```diff
@@ -299,7 +331,7 @@ def _get_company_unique_token_from_token(token):
     return get_object_or_404(CompanyUniqueToken, token=token)
 
 
-@permission_required('users_app.can_approve', raise_exception=True)
+@permission_required(CAN_APPROVE_PERM, raise_exception=True)
```
Comment (leandro-eb) [inline supplier_app/views.py]: I think having `CAN_APPROVE_PERM` without context it's not a good approach, consider providing context example: `COMPANY_USER_CAN_APPROVE_PERM`

File: users_app/tests/test_permissions.py [hunk 4]
```diff
@@ -186,3 +201,150 @@ def test_add_user_to_group_in_StrategyReporterGroup_class(self):
...
+
+        self.file_mock = MagicMock(spec=File)
+        self.file_mock.name = 'test.pdf'
+        self.file_mock.size = 50
+
+        self.company1 = CompanyFactory(
+            name='FakeCompany',
+            description='Best catering worldwide'
+        )
+        self.companyuserpermission1 = CompanyUserPermissionFactory(
```
Comment (leandro-eb) [inline users_app/tests/test_permissions.py]: Consider renaming with: `self.company_user_permission_1`

File: supplier_app/tests/test_view.py [hunk 5]
```diff
@@ -913,54 +939,61 @@ def test_GET_edit_address_view(self):
...
-                kwargs=self.kwargs_taxpayer
+                kwargs=self.kwargs_taxpayer_id,
             ),
             response.url
         )
 
-    def test_post_edit_address_info_as_supplier_redirects_to_supplier_home(self):
-        request = self.factory.post(
+    def test_post_edit_address_info_as_supplier_redirects_to_supplier_details(self):
+        response = self.client2.post(
```
Comment (leandro-eb) [inline supplier_app/tests/test_view.py]: Remember what we talked on our Code Review Session about having all specific resources tied to the unit test.

File: supplier_app/tests/test_view.py [hunk 6]
```diff
@@ -913,54 +939,61 @@ def test_GET_edit_address_view(self):
...
+        self.ap_user.groups.add(self.ap_group)
+        self.client.force_login(self.ap_user)
+
+        self.supplier_group = Group.objects.get(name="supplier")
+        self.supplier_user = UserFactory(email='nahuelSupplier@gmail.com')
+        self.supplier_user.groups.add(self.supplier_group)
+        self.client2.force_login(self.supplier_user)
 
         self.taxpayer = TaxPayerArgentinaFactory()
+        self.comapnyuserpermission = CompanyUserPermissionFactory(
```
Comment (gmarchetta-eb) [inline supplier_app/tests/test_view.py]: Same comment as https://github.com/evbeda/suppliers_management/pull/154/files#r338690477

File: supplier_app/views.py (hunk 2 above)
Comment (NahuelValencia-eb) [reply supplier_app/views.py]: Done

File: supplier_app/views.py (hunk 3 above)
Comment (NahuelValencia-eb) [reply supplier_app/views.py]: Done

File: users_app/tests/test_permissions.py (hunk 4 above)
Comment (NahuelValencia-eb) [reply users_app/tests/test_permissions.py]: done

File: supplier_app/tests/test_view.py (hunk 1 above)
Comment (NahuelValencia-eb) [reply supplier_app/tests/test_view.py]: CompanyUserPermission is the name of the table that we created at first. But now we are using the table of Django's permissions.

File: supplier_app/tests/test_view.py (hunk 5 above)
Comment (NahuelValencia-eb) [reply supplier_app/tests/test_view.py]: done

File: supplier_app/tests/test_view.py (hunk 6 above)
Comment (NahuelValencia-eb) [reply supplier_app/tests/test_view.py]: done

**Pull request merged**